pandas/indexes/base.py

```diff
@@ -67,6 +67,8 @@ def _new_Index(cls, d):
     """ This is called upon unpickling, rather than the default which doesn't
     have arguments and breaks __new__
     """
+    if issubclass(cls, ABCPeriodIndex):
```

Comment:
This change would do away with this stuff: https://github.com/SixtyCapital/pandas/compare/period-float...SixtyCapital:use-getstate